packages/embark-blockchain-process/src/baseBlockchainPlugin.ts

```diff
@@ -0,0 +1,100 @@
+import { Embark, Events } from "embark";
+import Web3 from "web3";
+import { BaseBlockchainClientOptions } from "../index";
+
+const NOT_IMPLEMENTED_EXCEPTION = "This method has not been implemented";
+
+/**
+ * Base blockchain plugin. Extend this class in your custom blockchain client plugin.
+ */
+export class BaseBlockchainPlugin {
```

Comment:
I'm not sure of the use of this base class. It looks like it's almost only for the account creation and unlocking in the case of not having a CLI command to do it, so I'm wondering if many other node clients will need it?

I'm asking because if in the end, almost only Lightchain needs `"blockchain:provider:ready"` and `createAndUnlockAccounts`, implementing this class for other plugins will be confusing.

Reply:
Creating and unlocking accounts (node account management) seems to be something that needs to happen for all blockchains, yet it's not something embark can abstract out due to the differences in each blockchain's methodology (unless we use RPC calls via `web3`, but what if it's not Ethereum?), so it seemed the best way forward was to let blockchains handle the account management themselves.

Firstly, a blockchain plugin needs to register itself in Embark, which can be done simply by creating a plugin that extends `BaseBlockchainPlugin`. The next step for the developer is to expose the method `createAndUnlockAccounts()`. If the dev doesn't want to create/unlock accounts, or the node doesn't support it, they can simply return from the overridden method.

To make this less confusing, we could remove the thrown error in the parent class for `createAndUnlockAccounts` and make this method an optional override. This would make the overall implementation of a blockchain plugin super simple:
```
import {BaseBlockchainPlugin} from 'embark-blockchain-process';
const BLOCKCHAIN_CLIENT_OPTIONS = {
  //...
};

export default class MyBlockchain extends BaseBlockchainPlugin {
  constructor(embark) {
    super(embark, BLOCKCHAIN_CLIENT_OPTIONS);
  }
}
```
What do you think?

Reply:
Yeah, I think making the override optional would be a good idea here